include/pstl_offload/internal/usm_memory_replacement_common.h

```diff
@@ -124,6 +145,12 @@ __realloc_real_pointer(void* __user_ptr, std::size_t __new_size)
     assert(__user_ptr != nullptr);
     __block_header* __header = static_cast<__block_header*>(__user_ptr) - 1;
 
+    if (!__new_size)
+    {
+        free(__user_ptr);
+        return nullptr;
+    }
+
```

Comment:
It seems like we can move the new code before calculating the header address.

Reply:
Done.